src/components/dropdown.component.js

```diff
@@ -1,43 +1,100 @@
-import $ from 'jquery';
-
-export default class DropdownComponent {
-    constructor(query) {
-        this.query = query;
-
-        this.btn = $(`${query} .dropdown-btn`);
-        this.items = $(`${query} .dropdown-item`);
-        this.list = $(`${query} .dropdown-list`);
-        this.selectedItem = $(`${query} .selected`);
-
-        this.addDropdownClickEventListner();
-        this.addItemClickEventListner();
-    }
-
-    addDropdownClickEventListner() {
-        this.btn.on('click', () => {
-            this.list.toggle();
-        });
-    }
-
-    addItemClickEventListner() {
-        this.items.on('click', (event) => {
-            this.unselectItem();
-            this.selectItem(event.currentTarget);
-            this.list.toggle();
-        });
-    }
-
-    updateDropdown() {
-        this.btn.html(this.selectedItem.text());
-    }
-
-    unselectItem() {
-        this.items.removeClass('selected');
-    }
-
-    selectItem(item) {
-        this.selectedItem = $(item);
-        this.selectedItem.addClass('selected');
-        this.updateDropdown();
-    }
-}
+import $ from 'jquery';
+import DropdownOptionNotFound from '../shared/erros/dropdow-option-not-found.erros';
+
+export default class DropdownComponent {
+    constructor(query, required = true) {
+        this.query = query;
+        this.required = required;
+        this.disabled = false;
+        this.valid = false;
+
+        this._container = $(`${query} .dropdown-container`);
+        this._btn = $(`${query} .dropdown-btn`);
+        this._list = $(`${query} .dropdown-list`);
+        this._items = $(`${query} .dropdown-item`);
+        this._currentSelected = $(`${query} .selected`);
+
+        this.selected = this._currentSelected.text();
+        this.defaultText = this._btn.text();

```

Comment:
This property must be a placeholder, if you want to use it must keep private
![image](https://user-images.githubusercontent.com/23269934/163377909-5e05ae39-d895-46d5-b235-11be65685579.png)

Reply:
I've set this private, because i use this as placeholder if it was an input tag